signature/cose/envelope.go

```diff
@@ -104,18 +105,18 @@ func (signer *remoteSigner) CertificateChain() []*x509.Certificate {
 	return signer.certs
 }
 
-// localSigner implements signer interface.
-// It is used in Sign process when go-cose's built-in signer is desired.
 type localSigner struct {
-	base         signature.LocalSigner
-	alg          cose.Algorithm
-	cryptoSigner crypto.Signer
-	certs        []*x509.Certificate
+	coseSigner cose.Signer
```

Comment:
How about
```suggestion
	cose.Signer
```
Then we don't even need to implement `Algorithm()` and `Sign(rand io.Reader, payload []byte)`.

Reply:
Good idea! Updated the code.